crates/core/src/kvs/ds.rs

```diff
@@ -680,36 +685,90 @@ impl Datastore {
 		Ok(())
 	}
 
-	/// Run the background task to perform changefeed garbage collection
+	/// Performs changefeed garbage collection as a background task.
+	///
+	/// This method is responsible for cleaning up old changefeed data across all databases.
+	/// It uses a distributed task lease mechanism to ensure that only one node in a cluster
+	/// performs this maintenance operation at a time, preventing duplicate work and potential
+	/// conflicts.
+	///
+	/// The process involves:
+	/// 1. Acquiring a lease for the ChangeFeedCleanup task
+	/// 2. Calculating the current system time
+	/// 3. Saving timestamps for current versionstamps
+	/// 4. Cleaning up old changefeed data from all databases
+	///
+	/// # Parameters
+	/// * `delay` - Duration specifying how long the lease should be valid
```

Comment:
Does this imply we need to carefulll set this `delay` to be longer than time any single GC run can take? Like, if the operator of the database thinks it will ever take up to 1h for GC at any scale, set this delay 1h so that no lease expire while a GC is still running on a node?

If so, that looks ok for now!

An alternative approach would be to have a "renewable" lease that can expire after `delay` ONLY IF the time since last renewable exceeds the configured delay. In that case, `delay` can be much less value like 1m or 2m, depending on how often you try to renew the lease.